File: grida-domains/domain/src/main/java/org/grida/domain/diary/DiaryRepository.java
package org.grida.domain.diary;

import org.grida.datetime.DateTimeRange;

import java.time.LocalDateTime;
import java.util.List;

public interface DiaryRepository {
    long save(String userEmail, DiaryContents contents, int imageRefreshChance, LocalDateTime lastActionAt);
    Diary findById(long id);
    List<Long> findIdsByUserEmailAndTargetDateBetween(String userEmail, DateTimeRange range);
    List<Diary> findAllByUseEmailAndDiaryCursor(String userEmail, DiaryCursor cursor);
    int findImageRefreshChanceById(long id);
    long modifyContents(long id, DiaryContents contents, LocalDateTime lastActionAt);
    void useImageRefreshChance(long id, LocalDateTime lastActionAt);
    void delete(long id);
}

File: grida-core/core-domain/src/main/kotlin/org/grida/domain/diary/DiaryRepository.kt
package org.grida.domain.diary

import org.grida.datetime.DateTimeRange
import org.grida.domain.user.User
import java.time.LocalDate

interface DiaryRepository {

    fun save(diary: Diary, user: User): Long

    fun findById(id: Long): Diary

    fun findAllByUserIdAndTargetDateBetween(userId: Long, range: DateTimeRange): List<Diary>

    fun existsByUserIdAndTargetDate(userId: Long, targetDate: LocalDate): Boolean

    fun updateContent(diaryId: Long, content: String): Long

    fun updateScope(diaryId: Long, scope: DiaryScope): Long
}
